server/lib/settings-location-update.js

```diff
@@ -0,0 +1,33 @@
+import nodeGeocoder from 'node-geocoder'
+import {fieldTypes} from '../../common/constants'
+import {getFieldsByType} from '../lib/questionnaire-helpers'
+import locationSchema from '../models/location-schema'
+import Settings from '../models/settings'
+
+// Initialize geocoder options for pre save method
+const geocoder = nodeGeocoder({
+  provider: 'google',
+  formatter: null
+})
+
+/**
+ * Hook a method to construct the geolocation of the address
+ */
+export async function updateLocationSettings(id){
+
+  const query = Settings.findById(id)
+  const settings = await query.select('+gmapsApiKey +gmapsClientId')
+
+  const address = settings.address
+
+  const [result] = await geocoder.geocode(address)
+
+  if (!result) return next(new ValidationError({
+    fields: Object.assign({}, ...addressFields.map((field, i) => ({
```

Comment:
The settings model doesn't have a 'fields' key, what key should the error be on in this case?